File: frontend/src/types/taskRun.ts
/** Statistiques extraites des logs JSON rclone (dernière ligne "stats") */
export interface RcloneStats {
  bytes: number;
  checks: number;
  deletedDirs: number;
  deletes: number;
  elapsedTime: number;
  errors: number;
  fatalError: boolean;
  renames: number;
  retryError: boolean;
  speed: number;
  totalBytes: number;
  totalChecks: number;
  totalTransfers: number;
  transferTime: number;
  transfers: number;
}

export interface TaskRunSummary {
  id: string;
  task_id: string;
  triggered_by: 'scheduler' | 'manual' | 'restore';
  status: 'running' | 'success' | 'failure';
  started_at: string;
  finished_at: string | null;
  duration_ms: number | null;
  exit_code: number | null;
  stats: RcloneStats | null;
}

export interface TaskRun extends TaskRunSummary {
  log_bytes?: number;
}

export interface RunLogLine {
  offset: number;
  text: string;
}

export interface RunLogPage {
  lines: RunLogLine[];
  total_bytes: number;
  at_start: boolean;
  at_end: boolean;
  prev_offset: number | null;
  next_offset: number | null;
}

export type RunLogQuery =
  | { from: 'start' | 'end' }
  | { after: number }
  | { before: number };

File: backend/src/models/task_run.rs
use serde::Serialize;
use uuid::Uuid;

/// Vue sans log_output pour les listes (inclut stats pour le tableau)
#[derive(Debug, Serialize)]
pub struct TaskRunSummary {
    pub id: Uuid,
    pub task_id: Uuid,
    pub triggered_by: String,
    pub status: String,
    pub started_at: chrono::DateTime<chrono::Utc>,
    pub finished_at: Option<chrono::DateTime<chrono::Utc>>,
    pub duration_ms: Option<i64>,
    pub exit_code: Option<i32>,
    pub stats: Option<serde_json::Value>,
}

/// Run metadata without the log blob.
#[derive(Debug, Serialize)]
pub struct TaskRunDetail {
    pub id: Uuid,
    pub task_id: Uuid,
    pub triggered_by: String,
    pub status: String,
    pub started_at: chrono::DateTime<chrono::Utc>,
    pub finished_at: Option<chrono::DateTime<chrono::Utc>>,
    pub duration_ms: Option<i64>,
    pub exit_code: Option<i32>,
    pub stats: Option<serde_json::Value>,
    pub log_bytes: i64,
}

#[derive(Debug, Serialize)]
pub struct LogLineView {
    pub offset: usize,
    pub text: String,
}

#[derive(Debug, Serialize)]
pub struct RunLogPage {
    pub lines: Vec<LogLineView>,
    pub total_bytes: usize,
    pub at_start: bool,
    pub at_end: bool,
    pub prev_offset: Option<usize>,
    pub next_offset: Option<usize>,
}
